### PR TITLE
bump MPFR to 3.1.6

### PR DESCRIPTION
**Pull request by simonbyrne**
We can also remove the patch from #22857.

### CODE REVIEW

**Review by ararslan — APPROVED** [review]

**Comment by simonbyrne** [comment]
The x86_64 Appveyor build seems to be timing out.

File: test/mpfr.jl [hunk 1]
```diff
@@ -864,11 +864,10 @@ end
...
     end
 end
+
 # issue #22758
-if MPFR.version() > v"3.1.5" || "r11590" in MPFR.patches()
-    setprecision(2_000_000) do
-        @test abs(sin(big(pi)/6) - 0.5) < ldexp(big(1.0),-1_999_000)
-    end
+setprecision(2_000_000) do
+    @test abs(sin(big(pi)/6) - 0.5) < ldexp(big(1.0),-1_999_000)
```
Comment (nalimilan) [inline test/mpfr.jl]: This test fails randomly when building Fedora nightlies with system mpfr 3.1.5-3 (see at the end of the log, hard to miss...):
https://copr-be.cloud.fedoraproject.org/results/nalimilan/julia-nightlies/fedora-26-i386/00700417-julia/build.log.gz

Any ideas?

Comment (simonbyrne) [reply test/mpfr.jl]: Ah. We can add the version check back. I removed it because I wasn't sure what the convention was.

Comment (nalimilan) [reply test/mpfr.jl]: OK. As long as we don't require 3.1.6 (which would be a bit extreme), we should definitely keep the check. See https://github.com/JuliaLang/julia/pull/25538.

**Pull request merged**